### PR TITLE
feat: add Vault integration Quarkus

### PR DESCRIPTION
**Pull request by CCH0124**
1. 新增 Vault KV engine 使用和與 Quarkus 框架整合

### CODE REVIEW

**Review by hwchiu — APPROVED** [review]
LGTM

File: docs/self-paced-labs/vault/index.md [hunk 1]
```diff
@@ -0,0 +1,667 @@
+---
+description: ""
+---
+
+# Quarkus 整合 Vault KV 引擎
+:::note
+本篇文章內容與實驗會同步[CCH0124 - vault-demo](https://github.com/CCH0124/quarkus-demo/tree/main/vault-demo/example/vault)
+:::
+
+HashiCorp Vault 屬於 CNCF 的 Key Management 類別的一員。
```
Comment (ydFu) [inline docs/self-paced-labs/vault/index.md]:
```suggestion
HashiCorp Vault 屬於 CNCF 的 [Key Management](https://landscape.cncf.io/card-mode?category=key-management&grouping=category) 類別的一員。
```

File: docs/self-paced-labs/vault/index.md [hunk 2]
```diff
@@ -0,0 +1,667 @@
+---
+description: ""
+---
+
+# Quarkus 整合 Vault KV 引擎
+:::note
+本篇文章內容與實驗會同步[CCH0124 - vault-demo](https://github.com/CCH0124/quarkus-demo/tree/main/vault-demo/example/vault)
```
Comment (ydFu) [inline docs/self-paced-labs/vault/index.md]:
```suggestion
本篇文章內容與實驗會同步 [CCH0124 - vault-demo](https://github.com/CCH0124/quarkus-demo/tree/main/vault-demo/example/vault)
```

File: docs/self-paced-labs/vault/index.md [hunk 3]
```diff
@@ -0,0 +1,667 @@
...
+
+
+## 實驗環境
+
+- OS: Windows 11 WSL
+- Docker Engine: 23.0.5
+- k3d version: v5.4.9
+- kubectl version: v1.27.1
+- Helm
+- Vault
```
Comment (ydFu) [inline docs/self-paced-labs/vault/index.md]:
```suggestion
- Helm
- Vault 
```
Miss Version

File: docs/self-paced-labs/vault/index.md [hunk 4]
```diff
@@ -0,0 +1,667 @@
...
+- Helm
+- Vault 
+
+## 環境安裝與配置
+1. [install docker](https://docs.docker.com/engine/install/ubuntu/)
+2. [install kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/#install-kubectl-on-linux)
+3. [install k3d](https://k3d.io/v5.4.9/#install-script)
+
+
+安裝完 k3d 後，使用下面 K3d 配置檔案建立一個 Kubernetes 環境
```
Comment (ydFu) [inline docs/self-paced-labs/vault/index.md]:
```suggestion
安裝完 k3d 後，使用下面 k3d 配置檔案建立一個 Kubernetes 環境：
```

File: docs/self-paced-labs/vault/index.md [hunk 5]
```diff
@@ -0,0 +1,667 @@
...
+當 Kubernetes 環境被建立之後，透過 `kubectl config` 指令可以看到被建立的環境如下
+```bash
+$ kubectl config get-contexts
+CURRENT   NAME                                                                 CLUSTER                                                              AUTHINFO                                                             NAMESPACE
+*         k3d-vault-cluster                                                    k3d-vault-cluster                                                    admin@k3d-vault-cluster
+```
+
+
+4. [Install Helm](https://helm.sh/docs/intro/install/#from-apt-debianubuntu)
+5. Install Nginx ingress controller
```
Comment (ydFu) [inline docs/self-paced-labs/vault/index.md]:
```suggestion
5. Install Nginx ingress controller:
```

File: docs/self-paced-labs/vault/index.md [hunk 6]
```diff
@@ -0,0 +1,667 @@
...
+
+在 `k3d` 上安裝 Kubernetes 時關閉了 `traefik` 選項，所以這邊用 `ingress-nginx` 取代，並使用 Helm chart 安裝
```
Comment (ydFu) [inline docs/self-paced-labs/vault/index.md]:
```suggestion
在 `k3d` 上安裝 Kubernetes 時關閉了 `traefik` 選項，所以這邊用 `ingress-nginx` 取代，並使用 Helm chart 安裝，安裝步驟如下：
```

File: docs/self-paced-labs/vault/index.md [hunk 7]
```diff
@@ -0,0 +1,667 @@
...
+service/ingress-nginx-controller             LoadBalancer   10.43.236.109   172.19.0.2,172.19.0.3,172.19.0.4   80:32369/TCP,443:31027/TCP   2m36s
+
+NAME                                       READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/ingress-nginx-controller   1/1     1            1           2m36s
+
+NAME                                                  DESIRED   CURRENT   READY   AGE
+replicaset.apps/ingress-nginx-controller-7d5fb757db   1         1         1       2m36s
+```
+
+6. Install Vault
```
Comment (ydFu) [inline docs/self-paced-labs/vault/index.md]:
```suggestion
6. Install Vault:
```

File: docs/self-paced-labs/vault/index.md [hunk 8]
```diff
@@ -0,0 +1,667 @@
...
+
+```bash
+helm repo add hashicorp https://helm.releases.hashicorp.com
+helm search repo hashicorp/vault
+quarkus-demo/vault-demo$ helm install vault hashicorp/vault --version 0.25.0 --namespace vault --create-namespace -f values/vault/values.yaml
+```
+
+關於 Vault 的 `values.yaml` 安裝配置可參考我的[鏈結](https://raw.githubusercontent.com/CCH0124/quarkus-demo/main/vault-demo/values/vault/values.yaml)
+
+安裝完之後必須作初始化和解封的動作
```
Comment (ydFu) [inline docs/self-paced-labs/vault/index.md]:
```suggestion
安裝完之後必須作初始化和解封的動作。
```

File: docs/self-paced-labs/vault/index.md [hunk 9]
```diff
@@ -0,0 +1,667 @@
...
+# 下面為推送的鍵值對
+======== Data ========
+Key              Value
+---              -----
+mqtt.broker      tcp://localhost:8883
+mqtt.password    demo1234
+mqtt.username    demo
+```
+
+如果資料定義錯誤想要做修正可以使用 `kv patch` 指令，假設 `mqtt.password` 設定錯誤要換成 demo12345678。這時又對資料進行了異動因此 `version` 加了一個版號
```
Comment (ydFu) [inline docs/self-paced-labs/vault/index.md]:
```suggestion
如果資料定義錯誤想要做修正可以使用 `kv patch` 指令，假設 `mqtt.password` 設定錯誤要換成 demo12345678。這時又對資料進行了異動因此 `version` 加了一個版號。
```

**Review by ydFu — APPROVED** [review]
LGTM,provide some detailed adjustments

File: docs/self-paced-labs/vault/index.md (hunk 3 above)
Comment (CCH0124) [reply docs/self-paced-labs/vault/index.md]: 已經新增

Comment (ydFu) [reply docs/self-paced-labs/vault/index.md]: Good Job,thanks.

**Comment by ydFu** [comment]
LGTM,thanks @CCH0124

**Review by steveyiyo — APPROVED** [review]
LGTM! Thanks for the contribution!

**Comment by steveyiyo** [comment]
Btw, Can we close #39 once we merge this PR?

**Pull request merged**